src/network/networkctl.c

```diff
@@ -836,8 +836,8 @@ static int list_links(int argc, char *argv[], void *userdata) {
                 (void) sd_network_link_get_operational_state(links[i].ifindex, &operational_state);
                 operational_state_to_color(links[i].name, operational_state, &on_color_operational, NULL);
 
-                r = sd_network_link_get_setup_state(links[i].ifindex, &setup_state);
-                if (r == -ENODATA) /* If there's no info available about this iface, it's unmanaged by networkd */
+                if (sd_network_link_get_setup_state(links[i].ifindex, &setup_state) < 0)
+                        /* If there's no info available about this iface, it's unmanaged by networkd */
                         setup_state = strdup("unmanaged");
```

Comment:
wouldn't it be nicer, to instead change the `table_add_many()` call below to pass `setup_state ?: "unmanaged` instead of just `setup_state`?

And then cast this `sd_network_link_get_setup_state()` call here to `(void)`, to explicitly ignore failures?

Reply:
(or maybe catch -ENOMEM explicitly, as for `net_get_type_string()` below)